pgxv4/settings.go

```diff
@@ -0,0 +1,76 @@
+//go:build go1.16
+// +build go1.16
+
+package pgxv4
+
+import (
+	"github.com/jackc/pgx/v4"
+
+	"github.com/avito-tech/go-transaction-manager/trm"
+)
+
+// Opt is a type to configure Settings.
+type Opt func(*Settings) error
+
+// WithTxOptions sets up pgx.TxOptions for the Settings.
+func WithTxOptions(opts *pgx.TxOptions) Opt {
+	return func(s *Settings) error {
+		*s = s.setTrOpts(opts)
+
+		return nil
+	}
+}
+
+// Settings contains settings for pgxv4.Transaction.
+type Settings struct {
+	trm.Settings
+	txOpts *pgx.TxOptions
```

Comment:
Why do you use pointer if pgx wants values?
Could lib just uses `pgx.TxOptions`?